holoviews/core/spaces.py

```diff
@@ -500,6 +517,8 @@ class DynamicMap(HoloMap):
        """)
 
     def __init__(self, callback, initial_items=None, **params):
+        if not isinstance(callback, Callable) and not isinstance(callback, types.GeneratorType):
+            callback = Callable(callable_function=callback)
```

Comment:
This means all callbacks will now be wrapped in ``Callable``. This is a change from the previous behavior, and probably a good one (more consistent, at least with ``__mull__``) though we should now document this. It would also be good to have an example of where a user might want to supply ``Callable`` themselves....